storage: clarify split stats handling

tbg opened the pull request:
Consume the incoming MVCCStats (which tracks writes to both sides)
and return one that contains a delta for the LHS only, along with
its sister delta for the RHS (contained in the PostCommitTrigger).
This avoids having to repurpose `deltaMS` half-way through.

<!-- Reviewable:start -->

---

This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/8055)

<!-- Reviewable:end -->

petermattis commented:
Review status: 0 of 1 files reviewed at latest revision, 3 unresolved discussions, all commit checks successful.

---

*[storage/replica_command.go, line 2535 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/8055#-KNg_CUCpoW9jl1OVHbQ:-KNg_CUCpoW9jl1OVHbR:194124945) ([raw file](https://github.com/cockroachdb/cockroach/blob/d0ce062bc3b3bc74a209264a33097fef349b5321/storage/replica_command.go#L2535)):*
> ```Go
> 	ctx context.Context,
> 	batch engine.Batch,
> 	bothDeltaMS enginepb.MVCCStats, // stats for batch so far
> ```

Perhaps I missed part of the previous discussion, but what was the problem with passing a pointer here. Given the single caller of `splitTrigger`, this doesn't seem to be a significant improvement.

---

*[storage/replica_command.go, line 2702 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/8055#-KNgZe11VeXe8LWLUUmd:-KNgZe11VeXe8LWLUUme:777766226) ([raw file](https://github.com/cockroachdb/cockroach/blob/d0ce062bc3b3bc74a209264a33097fef349b5321/storage/replica_command.go#L2702)):*
> ```Go
> 	// We've already recomputed that in absolute terms, so all we need to do is
> 	// to turn it into a delta so the upstream machinery can digest it.
> 	lhsDeltaMS := leftMS                  // start with new left-hand side absolute stats
> ```

For consistency: `s/lhs/left/g` and `s/rhs/right/g`

Or the opposite, though I prefer `left` and `right` because they are more easily distinguished.

Would also be nice to: `s/MS/Stats/g` (assuming this is confusing to other people).

---

*[storage/replica_command.go, line 2706 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/8055#-KNgZTlYNtFd6eYuNvxK:-KNgZTlYNtFd6eYuNvxL:226590175) ([raw file](https://github.com/cockroachdb/cockroach/blob/d0ce062bc3b3bc74a209264a33097fef349b5321/storage/replica_command.go#L2706)):*
> ```Go
> 	lhsDeltaMS.ContainsEstimates = false  // if there were any, recomputation removed them
> 
> 	// Performa a similar computation for the right hand side. The difference
> ```

`s/Performa/Perform/g`

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/8055)*
<!-- Sent from Reviewable.io -->

tbg committed:
storage: clarify split stats handling

Consume the incoming MVCCStats (which tracks writes to both sides)
and return one that contains a delta for the LHS only, along with
its sister delta for the RHS (contained in the PostCommitTrigger).
This avoids having to repurpose `deltaMS` half-way through.

tschottdorf commented:
Review status: 0 of 1 files reviewed at latest revision, 3 unresolved discussions, all commit checks successful.

---

*[storage/replica_command.go, line 2535 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/8055#-KNg_CUCpoW9jl1OVHbQ:-KNg_jQGBIozSEerpMzU:-427227829) ([raw file](https://github.com/cockroachdb/cockroach/blob/d0ce062bc3b3bc74a209264a33097fef349b5321/storage/replica_command.go#L2535)):*
<details><summary><i>Previously, petermattis (Peter Mattis) wrote…</i></summary>
> Perhaps I missed part of the previous discussion, but what was the problem with passing a pointer here. Given the single caller of `splitTrigger`, this doesn't seem to be a significant improvement.
</details>
`deltaMS` as incoming tracks writes to both sides of the split in the current batch. After the call we track only the LHS. So updating-in-place as before means that for most of the invocation it meant the former, but then had to be updated to mean the latter. I personally find it much clearer this way.

---

*[storage/replica_command.go, line 2702 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/8055#-KNgZe11VeXe8LWLUUmd:-KNga2bVMGp82ntZVkA_:1777411732) ([raw file](https://github.com/cockroachdb/cockroach/blob/d0ce062bc3b3bc74a209264a33097fef349b5321/storage/replica_command.go#L2702)):*
> For consistency: s/lhs/left/g and s/rhs/right/g

Done.

> Would also be nice to: s/MS/Stats/g (assuming this is confusing to other people).

`ms` is our de-facto variable name for `enginepb.MVCCStats`. That doesn't mean it's worthwhile keeping (I would also prefer `delta` or `deltaStats` though it's longer), but that effort should be global (`git grep -cE 'ms .?(enginepb\.)?MVCCStats'` should catch most of them).

---

*[storage/replica_command.go, line 2706 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/8055#-KNgZTlYNtFd6eYuNvxK:-KNgak_6cPjdUPjFl18j:-499146009) ([raw file](https://github.com/cockroachdb/cockroach/blob/d0ce062bc3b3bc74a209264a33097fef349b5321/storage/replica_command.go#L2706)):*
<details><summary><i>Previously, petermattis (Peter Mattis) wrote…</i></summary>
> `s/Performa/Perform/g`
</details>
Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/8055)*
<!-- Sent from Reviewable.io -->

petermattis commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

---

Review status: 0 of 3 files reviewed at latest revision, 3 unresolved discussions, all commit checks successful.

---

*[storage/replica_command.go, line 2535 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/8055#-KNg_CUCpoW9jl1OVHbQ:-KNgyWqxEdFPfvwunnIB:-465122292) ([raw file](https://github.com/cockroachdb/cockroach/blob/d0ce062bc3b3bc74a209264a33097fef349b5321/storage/replica_command.go#L2535)):*
<details><summary><i>Previously, tschottdorf (Tobias Schottdorf) wrote…</i></summary>
> `deltaMS` as incoming tracks writes to both sides of the split in the current batch. After the call we track only the LHS. So updating-in-place as before means that for most of the invocation it meant the former, but then had to be updated to mean the latter. I personally find it much clearer this way.
</details>
Got it. An alternative to return `leftDeltaMS` would be to pass a `leftDeltaMS *enginepb.MVCCStats` parameter. That seems mildly more symmetric with the way we pass stats around to other methods and would make the change somewhat smaller. Your call.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/8055#-:-KNgyp-IEm_-tJdZXvfc:1417088721)*
<!-- Sent from Reviewable.io -->

bdarnell commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

---

Reviewed 3 of 3 files at r2.
Review status: all files reviewed at latest revision, 3 unresolved discussions, all commit checks successful.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/8055#-:-KNi-Soc_cfJ0I8p75RQ:1417088721)*
<!-- Sent from Reviewable.io -->

tschottdorf commented:
TFTRs!

---

Review status: all files reviewed at latest revision, 3 unresolved discussions, all commit checks successful.

---

*[storage/replica_command.go, line 2535 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/8055#-KNg_CUCpoW9jl1OVHbQ:-KNiW-zV2TS-XGtjtuJ8:647705797) ([raw file](https://github.com/cockroachdb/cockroach/blob/d0ce062bc3b3bc74a209264a33097fef349b5321/storage/replica_command.go#L2535)):*
<details><summary><i>Previously, petermattis (Peter Mattis) wrote…</i></summary>
> Got it. An alternative to return `leftDeltaMS` would be to pass a `leftDeltaMS *enginepb.MVCCStats` parameter. That seems mildly more symmetric with the way we pass stats around to other methods and would make the change somewhat smaller. Your call.
</details>
I'll leave as is for now, have an ever so slight preference and there's already another function that does it "my" way (`writeInitialStats`, not surprising since I wrote that as well).

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/8055#-:-KNiWI28v779ERkhT4_G:2125754307)*
<!-- Sent from Reviewable.io -->
